net/dns/manager_darwin.go

```diff
@@ -83,8 +87,56 @@ func (c *darwinConfigurator) SetDNS(cfg OSConfig) error {
 	return c.removeResolverFiles(func(domain string) bool { return !keep[domain] })
 }
 
+// GetBaseConfig returns the current OS DNS configuration, extracting it from /etc/resolv.conf.
+// We should really be using the SystemConfiguration framework to get this information, as this
+// is not a stable public API, and is provided mostly as a compatibility effort with Unix
+// tools. Apple might break this in the future. But honestly, parsing the output of `scutil --dns`
+// is *even more* likely to break in the future.
 func (c *darwinConfigurator) GetBaseConfig() (OSConfig, error) {
-	return OSConfig{}, ErrGetBaseConfigNotSupported
+	cfg := OSConfig{}
+	resolvConf, err := os.ReadFile("/etc/resolv.conf")
+	if err != nil {
+		c.logf("failed to read /etc/resolv.conf: %v", err)
+		return cfg, ErrGetBaseConfigNotSupported
+	}
+	lines := strings.Split(string(resolvConf), "\n")
```

Comment:
We have a `resolv.conf` parser in the tree already; would it be possible to use that (or extend it) instead?
https://github.com/tailscale/tailscale/blob/main/net/dns/resolvconffile/resolvconffile.go

Reply:
Done in #13351